File: examples/03-interactions/go/child_workflow.go
package interactions

import (
	"go.temporal.io/sdk/workflow"
)

// A parent fans out to child Workflows the same way it fans out to Activities:
// start every child before getting any, then collect. ExecuteChildWorkflow
// returns a ChildWorkflowFuture; pass a TaskQueue per child so each can run on
// its own Worker pool.
func ParentOrderWorkflow(ctx workflow.Context, orderID string) error {
	fraudCtx := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{TaskQueue: "fraud"})
	shipCtx := workflow.WithChildOptions(ctx, workflow.ChildWorkflowOptions{TaskQueue: "shipping"})

	fraud := workflow.ExecuteChildWorkflow(fraudCtx, FraudWorkflow, orderID)
	shipping := workflow.ExecuteChildWorkflow(shipCtx, ShippingWorkflow, orderID)

	var fraudDecision, shippingPlan string
	if err := fraud.Get(ctx, &fraudDecision); err != nil {
		return err
	}
	if err := shipping.Get(ctx, &shippingPlan); err != nil {
		return err
	}
	return nil
}

File: examples/03-interactions/python/child_workflow.py
import asyncio

from temporalio import workflow


# A parent fans out to child Workflows the same way it fans out to Activities:
# start every child before awaiting any, then asyncio.gather. start_child_workflow
# returns a handle whose .result() resolves when the child completes; pass a
# task_queue so each child can run on its own Worker pool.
@workflow.defn
class ParentOrderWorkflow:
    @workflow.run
    async def process(self, order_id: str) -> None:
        fraud = await workflow.start_child_workflow(
            FraudWorkflow.check, order_id, task_queue="fraud"
        )
        shipping = await workflow.start_child_workflow(
            ShippingWorkflow.plan, order_id, task_queue="shipping"
        )

        await asyncio.gather(fraud, shipping)
